unit_tests/conftest.py

```diff
@@ -142,3 +142,14 @@ def fixture_params(request: pytest.FixtureRequest):
 @pytest.fixture(scope='function', autouse=True)
 def fixture_cleanup_continuous_events_registry():
     ContinuousEventsRegistry().cleanup_registry()
+
+
+def pytest_sessionfinish():
```

Comment:
According to the pytest docs [1] it must have 2 following parameters:

```
pytest_sessionfinish(session, exitstatus)

    session – The pytest session object.
    exitstatus – The status which pytest will return to the system.
```

[1] https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_sessionfinish

Reply:
> According to the pytest docs [1] it must have 2 following parameters:
> 
> ```
> pytest_sessionfinish(session, exitstatus)
> 
>     session – The pytest session object.
>     exitstatus – The status which pytest will return to the system.
> ```
> 
> So, have you tested it that it works?
> 
> [1] https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_sessionfinish

pytest is forgiving for missing arguments in hooks. i.e. if you are not using them you don't have to specific they.
the hooks are being called by name, and arguments are passed by name/position if exists in the signature.
(I think it give pytest the flexibility to extend the parameter list for hooks if needed)

Reply:
@vponomaryov @fruch thank you for your review, Valerii and Israel! the unused parameters are a little bit annoying.  the flycheck plugin of my editor keep reminding me it's bad.. so i just removed them.

Reply:
Ok, if it is optional then let's leave it as is.